src/yunohost/settings.py

```diff
@@ -235,3 +243,44 @@ def _save_settings(settings, location=SETTINGS_PATH):
             settings_fd.write(result)
     except Exception as e:
         raise YunohostError('global_settings_cant_write_settings', reason=e)
+
+
+# Meant to be a dict of setting_name -> function to call
+post_change_hooks = {}
+
+
+def post_change_hook(setting_name):
+    def decorator(func):
+        assert setting_name in DEFAULTS.keys(), "The setting %s does not exists" % setting_name
+        assert setting_name not in post_change_hooks, "You can only register one post change hook per setting (in particular for %s)" % setting_name
+        post_change_hooks[setting_name] = func
+        return func
+    return decorator
+
+
+def trigger_post_change_hook(setting_name, old_value, new_value):
+    if setting_name not in post_change_hooks:
+        logger.debug("Nothing to do after changing setting %s" % setting_name)
+        return
+
+    f = post_change_hooks[setting_name]
+    f(old_value, new_value)
+
+
+# ===========================================
+#
+# Actions to trigger when changing a setting
+# You can define such an action with :
+#
+# @post_change_hook("your.setting.name")
+# def some_function_name(old_value, new_value):
+#     # Do some stuff
+#
+# ===========================================
+
+
+#@post_change_hook("example.int")
+#def myfunc(old_value, new_value):
+#    print("In hook")
```

Comment:
Shall we protect this function from triggerring another setting change ?

Reply:
uuuh, I guess, but how would you do that ?

I'm not sure that we really are going to implement setting triggers likely to trigger other setting changes ?

Reply:
With decorator dependencies are not obvious. We could have a map that remembers what setting is currently triggering a hook and check that before entering hook and remove from map on hook exit.

Reply:
Uuuuuh yes I guess it's technically doable, but I doubt that we're going to need this anytime soon ...

Really most of the hook are going to be : 

```
@post_change_hook("some.setting")
def myfunc(...):
    ... regen some configuration like nginx ...
```

If we ever do very complicated stuff in the future (again I doubt it é_è) we'll still be able to implement it later

Reply:
Ok let's see how much time it will live before you ought me a 🍺 😆 !

Reply:
I'm ready to fix merge conflict on #640 once this branch is merged.

Reply:
I will sure you pay a beer if I can even if we don't end up implementing this :stuck_out_tongue: